pkg/apis/admissionregistration/validation/validation.go

```diff
@@ -1250,17 +1286,17 @@ func validateFieldRef(fieldRef string, fldPath *field.Path) field.ErrorList {
 
 // statelessCELCompiler does not support variable composition (and thus is stateless). It should be used when
 // variable composition is not allowed, for example, when validating MatchConditions.
-var statelessCELCompiler = plugincel.NewCompiler(environment.MustBaseEnvSet(environment.DefaultCompatibilityVersion()))
+var statelessCELCompiler = plugincel.NewCompiler(environment.MustBaseEnvSetWithStrictCostConfig(environment.DefaultCompatibilityVersion()))
 
 func createCompiler(allowComposition bool) plugincel.Compiler {
 	if !allowComposition {
 		return statelessCELCompiler
 	}
-	compiler, err := plugincel.NewCompositedCompiler(environment.MustBaseEnvSet(environment.DefaultCompatibilityVersion()))
+	compiler, err := plugincel.NewCompositedCompiler(environment.MustBaseEnvSetWithStrictCostConfig(environment.DefaultCompatibilityVersion()))
 	if err != nil {
 		// should never happen, but cannot panic either.
 		utilruntime.HandleError(err)
-		return plugincel.NewCompiler(environment.MustBaseEnvSet(environment.DefaultCompatibilityVersion()))
+		return plugincel.NewCompiler(environment.MustBaseEnvSetWithStrictCostConfig(environment.DefaultCompatibilityVersion()))
 	}
 	return compiler
```

Comment:
Since we're now passing in a `strictCostEnforcement` option into validation, can we also wire that option into these compile calls?  (Now that we have two feature gates, letting `environment.MustBaseEnvSet*` read feature gates doesn't work out as well).

Reply:
I am not sure if I got the point. We are looping through all opts(params, authz, strictCost) under NewCompiler so we build the env with desired opts. `environment.MustBaseEnvSet*` didn't read feature gate though..